src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/shared/cdk/cdk_stacks/tests/test_default_pipeline_type.py

```diff
@@ -0,0 +1,244 @@
+    }
+
+    stack_input["ssm_params"][region_name] = {
+        "modules": "fake-bucket-name",
+        "kms": f"arn:aws:kms:{region_name}:{acount_id}:key/my-unique-kms-key-id",
+    }
+    app = core.App()
+    PipelineStack(app, stack_input)
+
+    cloud_assembly = app.synth()
+    resources = {
+        k[0:-8]: v for k, v in cloud_assembly.stacks[0].template["Resources"].items()
+    }
+    code_pipeline = resources["codepipeline"]
+    assert code_pipeline["Type"] == "AWS::CodePipeline::Pipeline"
+    assert len(code_pipeline["Properties"]["Stages"]) == 4
+
+    target_1_wave_1 = code_pipeline["Properties"]["Stages"][2]
+    assert target_1_wave_1["Name"] == "deployment-stage-1-wave-0"
+    assert len(target_1_wave_1["Actions"]) == 3
+
+    target_1_wave_2 = code_pipeline["Properties"]["Stages"][3]
+    assert target_1_wave_2["Name"] == "deployment-stage-1-wave-1"
+    assert len(target_1_wave_2["Actions"]) == 3
+
+
+
+def test_pipeline_creation_outputs_as_expected_when_input_has_2_targets_with_2_waves_and_1_wave():
+    region_name = "eu-central-1"
+    acount_id = "123456789012"
+
+    stack_input = {
+        "input": {"params": {}, "default_providers": {}, "regions": {}},
+        "ssm_params": {"fake-region": {}},
+    }
+
+    stack_input["input"]["name"] = "test-stack"
+
+    stack_input["input"]["default_providers"]["source"] = {
+        "provider": "s3",
+        "properties": {"account_id": "123456789012"},
+    }
+    stack_input["input"]["default_providers"]["build"] = {
+        "provider": "codebuild",
+        "properties": {"account_id": "123456789012"},
+    }
+
+    stack_input["ssm_params"][region_name] = {
+        "modules": "fake-bucket-name",
+        "kms": f"arn:aws:kms:{region_name}:{acount_id}:key/my-unique-kms-key-id",
+    }
+    app = core.App()
+    PipelineStack(app, stack_input)
+
+    cloud_assembly = app.synth()
+    resources = {
+        k[0:-8]: v for k, v in cloud_assembly.stacks[0].template["Resources"].items()
+    }
+    code_pipeline = resources["codepipeline"]
+    assert code_pipeline["Type"] == "AWS::CodePipeline::Pipeline"
+    assert len(code_pipeline["Properties"]["Stages"]) == 2
+
+    source_stage = code_pipeline["Properties"]["Stages"][0]
+    assert len(source_stage["Actions"]) == 1
+
+    source_stage_action = source_stage["Actions"][0]
+    assert source_stage_action["ActionTypeId"]["Category"] == "Source"
+    assert source_stage_action["ActionTypeId"]["Owner"] == "AWS"
+    assert source_stage_action["ActionTypeId"]["Provider"] == "S3"
+
+    build_stage = code_pipeline["Properties"]["Stages"][1]
+    build_stage_action = build_stage["Actions"][0]
+    assert build_stage_action["ActionTypeId"]["Category"] == "Build"
+    assert build_stage_action["ActionTypeId"]["Owner"] == "AWS"
+    assert build_stage_action["ActionTypeId"]["Provider"] == "CodeBuild"
+
+    assert len(build_stage["Actions"]) == 1
+
+    region_name = "eu-central-1"
```

Comment:
Same here, looks like the function def is missing. As another test setup kicks-off from here.

Reply:
Think this was just a bad copy paste job, it's the same test from the pipeline_creation test